Common/src/Common/Service/Table/Formatter/ExternalConversationLink.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Common\Service\Table\Formatter;
+
+use Common\Service\Helper\UrlHelperService;
+use Common\Util\Escape;
+use DateTimeImmutable;
+use DateTimeInterface;
+
+class ExternalConversationLink implements FormatterPluginManagerInterface
+{
+    private UrlHelperService $urlHelper;
+
+    public function __construct(UrlHelperService $urlHelper)
+    {
+        $this->urlHelper = $urlHelper;
+    }
+
+    /**
+     * status
+     *
+     * @param array $row Row data
+     * @param array $column Column data
+     *
+     * @return     string
+     * @inheritdoc
+     */
+    public function format($row, $column = null): string
+    {
+        $route = 'messages/view';
+        $params = [
+            'conversationId' => $row['id'],
+        ];
+
+        $statusCSS = '';
+
+        switch ($row['userContextStatus']) {
+            case "NEW_MESSAGE":
+                $statusCSS = 'govuk-!-font-weight-bold';
+                $tagColor = 'govuk-tag--red';
+                break;
+            case "OPEN":
+                $tagColor = 'govuk-tag--blue';
+                break;
+            case "CLOSED":
+                $tagColor = 'govuk-tag--grey';
+                break;
+            default:
+                $tagColor = 'govuk-tag--green';
+                break;
+        }
+
+        $rows = '<a class="' . 'govuk-body govuk-link govuk-!-padding-right-1 ' . $statusCSS . '" href="%s">%s: %s</a>
```

Comment:
Not one for right now, but perhaps to raise an an Improvement.. Could we re-use anything in Common/src/Common/View/Helper/ (maybe Link.php & Status.php) here? Would mean any future changes to govuk Markup need tweaking in less places.

Reply:
👍